### DISCUSSION TITLE
Trouble with running SOFA with python

### DISCUSSION THREAD

**Discussion by LuyuHuang-AO** [sofa-framework/sofa#3728] [Develop in C++] [answered]
Hi,

runSofa actually works fine for me. However, I am now having trouble to run simulations with python directly.

I am running SOFA 22.12 in Ubuntu 22.04, I have added SofaPython3 via in-tree build.
SOFA_ROOT = /path/to/SOFA/build/install
PYTHONPATH = /path/to/SOFA/build/lib/python3/site-packages:/opt/SofaPython3/build/lib/python3/site-packages:

Here is the error that I am running into:
Traceback (most recent call last):
  File "14-python3-basic.py", line 2, in <module>
    import Sofa
  File "/path/to/SOFA/build/lib/python3/site-packages/Sofa/__init__.py", line 107, in <module>
    import Sofa.Helper
ModuleNotFoundError: No module named 'Sofa.Helper'

Is there any insight you could provide to help with troubleshooting? Thank you!

**Comment by alxbilger** [comment, marked as answer]
What OS are you running?
I already had this error. It was because of two versions of Python on my computer. You need to run the Python interpreter corresponding to the version used to compile SofaPython3.

**Reply by LuyuHuang-AO** [reply]
You are right, I have both python3.10 and python3.8 installed. 
I tried running the script by specifying python3.10 and it worked! Thank you sir. 
Not sure why it does not work with python3.8, I thought it was supposed be the version we use for SofaPython3.

**Reply by alxbilger** [reply]
I think it's possible to compile SofaPython3 with Python 3.10.
